ui/src/shared/components/ViewSwitcher.tsx

```diff
@@ -186,4 +206,13 @@ const ViewSwitcher: FunctionComponent<Props> = ({
   }
 }
 
-export default ViewSwitcher
+const mstp = ({timeMachines}: AppState) => {
+  const {activeTimeMachineID, timeMachines: machine} = timeMachines
+  const {timeRange} = machine[activeTimeMachineID]
+  return {
+    startTime: timeRange ? getStartTime(timeRange) : Infinity,
+    endTime: timeRange ? getEndTime(timeRange) : null,
```

Comment:
it would be cleaner if the `getStartTime` and `getEndTime` functions handled returning infinity/null in cases where `timeRange` is undefined. I also have a feeling getEndTime should maybe return -Infinity? or startTime should default to `null`, it feels brittle to need to know that their defaults are different.